exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/window/DefaultFrameTemplate.java

```diff
@@ -287,6 +318,36 @@ private int countPeers(final int start) {
   }
 
   /**
+   * Counts how many rows are peer with the first row of the current frame. This is called when we don't require all
+   * batches of current partition to be processed at once.<br>
+   * Assumes the end of the frame has indeed been found, because of this it doesn't use partition.remaining to check
+   * the end of current partition but rather isSamePartition()
+   * @param start first row of current frame
+   * @return number of peer rows
+   */
+  private int countPeersPartialPartition(final int start) {
```

Comment:
This should be a long since the number of peers could easily be greater than Integer.MAX_VALUE.